config/formats.ts

```diff
@@ -2,7 +2,79 @@
 // The rules that formats use are stored in data/rulesets.js
 import {Utils} from './../lib/utils';
 
-export const Formats: (FormatsData | {section: string, column?: number})[] = [
+import {SideFormats} from './custom-formats';
+
+// interface for the builder.
+interface FormatSection {
+	section: string
+	column?: number
+	formats: FormatsData[];
+}
+
+// function for merging the two lists
+function merge(main: (FormatsData | {section: string, column?: number})[],
+			   side: (FormatsData | {section: string, column?: number})[]):
+			   (FormatsData | {section: string, column?: number})[] {
```

Comment:
```suggestion
function merge(
			   main: (FormatsData | {section: string, column?: number})[],
			   side: (FormatsData | {section: string, column?: number})[]
): (FormatsData | {section: string, column?: number})[] {
```
Also I feel like declaring a `type` for `FormatsData | {section: string, column?: number}` might be more readable

Reply:
I will add it.